### PR TITLE
Use RzOutputMode for `rz_core_file_print()` (`o` cmd)

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Just use `RzOutputMode` enum for the printing mode, also renamed the function to reflect it's true essence

**Test plan**

CI is green

**Closing issues**

Partially addresses https://github.com/rizinorg/rizin/issues/489 in preparation for https://github.com/rizinorg/rizin/issues/1589

### CODE REVIEW

File: librz/core/cfile.c [hunk 1]
```diff
@@ -1285,7 +1285,7 @@ RZ_API RzCoreFile *rz_core_file_get_by_fd(RzCore *core, int fd) {
 	return NULL;
 }
 
-RZ_API int rz_core_file_list(RzCore *core, int mode) {
+RZ_API int rz_core_file_print(RzCore *core, RzOutputMode mode) {
```
Comment (wargio) [inline librz/core/cfile.c]: how about converting this to bool ?

Comment (wargio) [reply librz/core/cfile.c]: or even better, the status one.

Comment (XVilka) [reply librz/core/cfile.c]: Converted to a bool.

File: librz/core/cfile.c [hunk 2]
```diff
@@ -1338,7 +1337,7 @@ RZ_API int rz_core_file_list(RzCore *core, int mode) {
 				free(absfile);
 			}
 			break;
-		case 'n': {
+		case RZ_OUTPUT_MODE_LONG: {
```
Comment (ret2libc) [inline librz/core/cfile.c]: this seems to be RZ_OUTPUT_MODE_RIZIN, not LONG

Comment (XVilka) [reply librz/core/cfile.c]: It's already been used before and the output is different.

Comment (ret2libc) [reply librz/core/cfile.c]: create a separate function for this. It's probably not even tested anyway, but it does not seem related to the other modes. THis is indeed a very separate thing, which deserves its own function.

Comment (XVilka) [reply librz/core/cfile.c]: Done

**Review by ret2libc — APPROVED** [review]
Good now!

**Pull request merged**